module/genericsUtils.ts

```diff
@@ -12,13 +12,24 @@ if ("BANANA_URL" in process.env){
 }
 
 
-export async function runMain(apiKey: string, modelKey: string, modelInputs: object = {}): Promise<object>{
-    const callID = await startAPI(apiKey, modelKey, modelInputs)
+export async function runMain(apiKey: string, modelKey: string, modelInputs: object = {}): Promise<any>{
+    const startOut = await startAPI(apiKey, modelKey, modelInputs)
+    if (startOut["finished"] == true){
```

Comment:
Did you check what happens if "finished" doesn't exist? will this crash?

Reply:
i.e assuming runMain is called for older sdks too can this ever not return the key properly

Reply:
if /start doesn't return it for some reason, the SDK falls back on our /check endpoint which doesn't use this "finished" logic as a break condition.